app/src/test/java/org/greenstand/android/TreeTracker/models/messages/MessagesRepoTest.kt

```diff
@@ -0,0 +1,203 @@
+package org.greenstand.android.TreeTracker.models.messages
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import io.mockk.coEvery
+import io.mockk.coVerify
+import io.mockk.mockk
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.TestCoroutineDispatcher
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.runBlockingTest
+import kotlinx.coroutines.test.setMain
+import org.greenstand.android.TreeTracker.MainCoroutineRule
+import org.greenstand.android.TreeTracker.models.UserRepo
+import org.greenstand.android.TreeTracker.models.messages.database.MessagesDAO
+import org.greenstand.android.TreeTracker.models.messages.database.entities.MessageEntity
+import org.greenstand.android.TreeTracker.models.messages.network.MessagesApiService
+import org.greenstand.android.TreeTracker.models.messages.network.responses.*
+import org.greenstand.android.TreeTracker.models.user.User
+import org.greenstand.android.TreeTracker.utilities.TimeProvider
+import org.junit.*
+import org.junit.rules.TestRule
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+@OptIn(ExperimentalCoroutinesApi::class)
+@RunWith(JUnit4::class)
+class MessagesRepoTest {
+
+    @get:Rule
+    val rule: TestRule = InstantTaskExecutorRule()
+
+    // Set the main coroutines dispatcher for unit testing.
+    @get:Rule
+    var mainCoroutineRule = MainCoroutineRule()
+
+    private lateinit var testDispatcher: TestCoroutineDispatcher
+
+    private lateinit var apiService: MessagesApiService
+    private lateinit var userRepo: UserRepo
+    private lateinit var timeProvider: TimeProvider
+    private lateinit var messagesDAO: MessagesDAO
+    private lateinit var messageUploader: MessageUploader
+
+    // system under test
+    private lateinit var messagesRepo: MessagesRepo
+
+    @Before
+    fun setUp() {
+        testDispatcher = TestCoroutineDispatcher()
+        Dispatchers.setMain(testDispatcher)
+
+        apiService = mockk()
+        userRepo = mockk()
+        timeProvider = mockk()
+        messagesDAO = mockk()
+        messageUploader = mockk()
+
+        messagesRepo = MessagesRepo(
+            apiService,
+            userRepo,
+            timeProvider,
+            messagesDAO,
+            messageUploader
+        )
+    }
+
+    @After
+    fun tearDown() {
+        Dispatchers.resetMain()
+    }
+
+    private val userList: List<User> = List(10) {
+        User(
+            it.toLong(),
+            it.toString(),
+            it,
+            it.toString(),
+            it.toString(),
+            it.toString(),
+            false,
+            it % 2 == 0
+        )
+    }
+
+    private val lastTimeMessageSynced = "123456789"
+
+    @Test
+    fun syncMessagesTest() = mainCoroutineRule.runBlockingTest {
```

Comment:
This method test name is vague, can we give it more detail?

Reply:
@Elforama  I have none coming in my mind, can you suggest one?